src/main/java/frc/trigon/robot/subsystems/climber/Climber.java

```diff
@@ -0,0 +1,120 @@
+package frc.trigon.robot.subsystems.climber;
+
+import com.ctre.phoenix6.controls.DynamicMotionMagicVoltage;
+import com.ctre.phoenix6.controls.VoltageOut;
+import edu.wpi.first.units.Measure;
+import edu.wpi.first.units.Units;
+import edu.wpi.first.units.Voltage;
+import edu.wpi.first.wpilibj.sysid.SysIdRoutineLog;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.trigon.robot.subsystems.MotorSubsystem;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXMotor;
+import org.trigon.hardware.phoenix6.talonfx.TalonFXSignal;
+
+public class Climber extends MotorSubsystem {
+    private final TalonFXMotor
+            rightMotor = ClimberConstants.RIGHT_MOTOR,
+            leftMotor = ClimberConstants.LEFT_MOTOR;
+    private final DynamicMotionMagicVoltage
+            nonClimbingPositionRequest = new DynamicMotionMagicVoltage(
+            0,
+            ClimberConstants.MAX_NON_CLIMBING_VELOCITY,
+            ClimberConstants.MAX_NON_CLIMBING_ACCELERATION,
+            0
+    ).withSlot(ClimberConstants.GROUNDED_SLOT).withEnableFOC(ClimberConstants.ENABLE_FOC),
+            climbingPositionRequest = new DynamicMotionMagicVoltage(
+                    0,
+                    ClimberConstants.MAX_CLIMBING_VELOCITY,
+                    ClimberConstants.MAX_CLIMBING_ACCELERATION,
+                    0
+            ).withSlot(ClimberConstants.ON_CHAIN_SLOT).withEnableFOC(ClimberConstants.ENABLE_FOC);
+    private final VoltageOut voltageRequest = new VoltageOut(0).withEnableFOC(ClimberConstants.ENABLE_FOC);
+    private ClimberConstants.ClimberState currentState = ClimberConstants.ClimberState.RESTING;
```

Comment:
It's the target state, not the current state because we haven't neccesarily reached it yet